### PR TITLE
Rule proposal: sort-imports

### PR DESCRIPTION
**Pull request by Betree**
We had a broken build on API because we ended up with a duplicate import after a merge (see https://github.com/opencollective/opencollective-api/pull/3609).

The [sort-imports](https://eslint.org/docs/rules/sort-imports) rule can prevent that: by having a predictable order for imports duplicates will normally result in an automatic merge or a git conflict.

**How can un-ordered imports result in a broken build**

`Base file`
```es6
import test1 from 'test1'
import test2 from 'test2'
```

`Branch 1`
```es6
import test3 from 'test3' // Add this
import test1 from 'test1'
import test2 from 'test2'
```

`Branch 2`
```es6
import test1 from 'test1'
import test2 from 'test2'
import test3 from 'test3'  // Add this
```

Merge `Branch 1` then `Branch 2` and you will end up with:
```es6
import test3 from 'test3' // Merged from Branch 1
import test1 from 'test1'
import test2 from 'test2'
import test3 from 'test3' // Merged from Branch 2

// => Babel error: duplicate test3 import
```

### CODE REVIEW

**Comment by Betree** [comment]
@kewitz sort-keys could be interesting too, tho the risk of having a conflict on object's keys (compared to the risk of having a conflict on import) seems lower to me. 

Also I like the fact that we can sort with other criterias, for example I like to put child objects at the end:
```es6
{
  zzz: 42,
  abc: {
    any: true
  }
}
```

That being said, I'm open to have this rule in place given that it can increase readability and code navigation - especially for large objects.

**Comment by znarf** [comment]
Maybe we should have a look at https://github.com/lydell/eslint-plugin-simple-import-sort

This is implementing more complex sort rules.

**Comment by znarf** [comment]
Or https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md

**Comment by Betree** [comment]
@znarf could you summarize what are the advantages of using `eslint-plugin-simple-import-sort` or `eslint-plugin-import`?

**Comment by znarf** [comment]
/ | sort-imports | import/order | simple-import-sort |
--- | ------------ | ------------ | ------------------ |
group imports by type |      N       |      Y       |         ?          |


wip ...

**Comment by kewitz** [comment]
Grouping is also desirable IMO, I think the most important thing is to have fix capability so we can just save and have auto-fix taking care of it.

**Comment by Betree** [comment]
I like `group imports by type (builtin, package, lib, ...)` and `sort on path` because it has been kind of implicit that we should do that in the code as of today so it will produce less changes. 

For the other rules I have no strong opinion, as long as the sort order is predictable either `import/order` or `simple-import-sort` will do the job if it's not too hard to fix the build.

**Comment by znarf** [comment]
There are some kind of circular imports that are super fragile in GraphQL v1, changing the order can break here. Need to find a work around.

**Comment by znarf** [comment]
The `simple-import-sort` https://github.com/opencollective/opencollective-api/pull/3621 PR is fixed. I think `import/order` could be similarly fixed.

**Comment by Betree** [comment]
Updated this PR to use `simple-import-sort`
